northstar/src/runtime/process/fork.rs

```diff
@@ -0,0 +1,24 @@
+use futures::Future;
+use nix::unistd::{self, Pid};
+
+use crate::runtime::error::Error;
+
+/// Fork a new process.
+///
+/// The child process runs the given closure and exits.
+///
+pub async fn fork_process<F>(f: F) -> nix::Result<Pid>
+where
+    F: Future<Output = Result<(), Error>>,
+{
+    match unsafe { unistd::fork()? } {
+        unistd::ForkResult::Parent { child } => Ok(child),
+        unistd::ForkResult::Child => {
+            if let Err(e) = f.await {
+                log::error!("Failed after fork: {:?}", e);
+                std::process::exit(-1);
+            }
```

Comment:
I thought using the tokio runtime here without spawning a new runtime in a thread is problematic

Reply:
@marcmo It is.
The current version of the patchset doesn't use any `async` code beyond the fork.